### PR TITLE
Add isotropic viscosity and resistivity

### PR DESCRIPTION
**Pull request by pgrete**
What is says on the can: adds support for isotropic viscosity and resistivity

Description on usage and implementation part of the PR itself (so won't repeat here).

The artifacts from the new regression tests are for the viscosity
visc
and for the resistivity
ohm

Should be ready for review and comments.

### CODE REVIEW

File: src/hydro/diffusion/viscosity.cpp [hunk 1]
```diff
@@ -0,0 +1,295 @@
+//========================================================================================
+// AthenaPK - a performance portable block structured AMR astrophysical MHD code.
+// Copyright (c) 2021-2023, Athena-Parthenon Collaboration. All rights reserved.
```
Comment (bwoshea) [inline src/hydro/diffusion/viscosity.cpp:3]: 2023 -> 2024

File: tst/regression/test_suites/diffusion/diffusion.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
+# ========================================================================================
+# AthenaPK - a performance portable block structured AMR MHD code
+# Copyright (c) 2023, Athena Parthenon Collaboration. All rights reserved.
```
Comment (bwoshea) [inline tst/regression/test_suites/diffusion/diffusion.py:3]: 2024, maybe?

File: tst/regression/test_suites/diffusion_linwave3d/diffusion_linwave3d.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
+# ========================================================================================
+# AthenaPK - a performance portable block structured AMR MHD code
+# Copyright (c) 2023, Athena Parthenon Collaboration. All rights reserved.
```
Comment (bwoshea) [inline tst/regression/test_suites/diffusion_linwave3d/diffusion_linwave3d.py:3]: 2023 -> 2024?

**Review by bwoshea — APPROVED** [review]
Overall this looks great - I only have minor comments.  Sorry it took me so long to go through it!

File: docs/input.md [hunk 4]
```diff
@@ -69,6 +69,62 @@ conserved to primitive conversion if both are defined.
 
 #### Diffusive processes
 
+Diffusive processes in AthenaPK can be configured in the `<diffusion>` block of the input file.
+```
+<diffusion>
+integrator = unsplit       # alternatively: rkl2
+#rkl2_max_dt_ratio = 100.0 # limits the ratio between the parabolic and hyperbolic timesteps
+#cfl = 1.0                 # Additional safety factor applied to diffusive timestep constraint. Default to hyperbolic cfl.
```
[the remark this answers is not in the document]
Comment (pgrete) [reply docs/input.md:77]: Actually this applies to both. I adjusted the comment accordingly.

File: src/hydro/diffusion/resistivity.cpp [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+  const auto ndim = prim_pack.GetNdim();
+
+  Real fac = 0.5;
+  if (ndim == 2) {
+    fac = 0.25;
+  } else if (ndim == 3) {
+    fac = 1.0 / 6.0;
+  }
+
+  const auto gm1 = hydro_pkg->Param<Real>("AdiabaticIndex");
```
[the remark this answers is not in the document]
Comment (pgrete) [reply src/hydro/diffusion/resistivity.cpp:52]: Not needed (leftover copy paste line) and removed.

File: docs/input.md [hunk 6]
```diff
@@ -140,6 +196,29 @@ Default value corresponds to the typical value used in literature and goes back
 [^BM82]:
     S. A. Balbus and C. F. McKee, “The evaporation of spherical clouds in a hot gas. III - Suprathermal evaporation,” , vol. 252, pp. 529–552, Jan. 1982, doi: https://doi.org/10.1086/159581
 
+#### Viscosity/Momentum diffusion
+
+Only isotropic viscosity with a (spatially and temporally) fixed coefficient in code units
+is currently implemented.
```
[the remark this answers is not in the document]
Comment (pgrete) [reply docs/input.md:202]: Added info both here and in the overview block above.

File: docs/input.md [hunk 7]
```diff
@@ -69,6 +69,62 @@ conserved to primitive conversion if both are defined.
...
+Diffusive processes in AthenaPK can be configured in the `<diffusion>` block of the input file.
+```
+<diffusion>
+integrator = unsplit       # alternatively: rkl2
+#rkl2_max_dt_ratio = 100.0 # limits the ratio between the parabolic and hyperbolic timesteps
+#cfl = 1.0                 # Additional safety factor applied to diffusive timestep constraint. Default to hyperbolic cfl.
+
+conduction = anisotropic               # none (disabled), or isotropic, or anisotropic
+conduction_coeff = fixed               # alternative: spitzer
+thermal_diff_coeff_code = 0.01         # fixed coefficent in code units
```
[the remark this answers is not in the document]
Comment (pgrete) [reply docs/input.md:81]: I just added the info on the units itself.
Regarding text book references, I'm happy if any student who does a deep dive intro opens a PR to extend the documentation accordingly.

**Pull request merged**